src/ddmd/parse.d

```diff
@@ -5845,7 +5845,7 @@ final class Parser(AST) : Lexer
                 if (token.value == TOKfinally)
                 {
                     nextToken();
-                    finalbody = parseStatement(0);
+                    finalbody = parseStatement(PSscope);
```

Comment:
> The fix for issue 17690 is here: #7026

Thanks! It helps to put a link to this PR in bugzilla. That way, nobody is going to waste time developing a fix when one already exists.